app/services/user_authentication/user_set_config.py

```diff
@@ -0,0 +1,23 @@
+import os
+import shutil
+
+from app.configs import app_config
+from app.services.logger_services import log_functions as logger
+from app.services.output_manager.error_handler import ECustomizedError
+from app.services.output_manager.error_handler import SrvErrorHandler
+
+
+def check_config():
+    connections = app_config.AppConfig.Connections.__dict__
+    for k, v in connections.items():
+        if k.startswith('url') and not v:
```

Comment:
if you are checking for `None` you should explicitly state `and is None`.

It's more idiomatic, technically more correct as there is exist values that would make the value falsy and also make the condition easier to read.

Reply:
I think there are two cases here:
1. user does not have config file, so value is None
2. user has config but cleared the value, so value is ' ' (empty string)
I think use 'is None' can only handle case 1, while use 'not' could handle both cases.

Reply:
Are you not calling the following when true?
```python
SrvErrorHandler.customized_handle(ECustomizedError.NO_CONFIG_FILE, True)
```

Case 2 is. not a valid precondition for that error message. The other solution is to return a different error.

Reply:
No, it won't calling that function when value is true. The function check_config is only used in a decorator for each command group except use_config, and it will print proper formatted message and stop execution with sys.exit.
For case 2. I have updated the variable to CONFIG_NOT_FOUND, the corresponding error message in the app/resources/custom_error.py will be reviewed by ops team later.